internal/runner/runnerpool/runner.go

```diff
@@ -399,3 +631,69 @@ func (r *Runner) SetParseOptions(parserOptions []hclparse.Option) {
 func (r *Runner) SetReport(report *report.Report) {
 	r.Stack.Report = report
 }
+
+// SetUnitFilterCallback sets the unit filter callback.
+func (r *Runner) SetUnitFilterCallback(callback common.UnitFilterCallback) {
+	r.unitFilterCallback = callback
+}
+
+// isDestroyCommand checks if the current command is a destroy operation
+func isDestroyCommand(opts *options.TerragruntOptions) bool {
```

Comment:
This doesn't catch `apply -destroy`. I know we have logic somewhere to rewrite this, but I think we should be explicit about that logic here.